Api/csharp/ArmoniK.Api.Client/Submitter/ReadOnlyByteMemoryPayload.cs

```diff
@@ -0,0 +1,59 @@
+// This file is part of the ArmoniK project
+// 
+// Copyright (C) ANEO, 2021-2022. All rights reserved.
+//   W. Kirschenmann   <wkirschenmann@aneo.fr>
+//   J. Gurhem         <jgurhem@aneo.fr>
+//   D. Dubuc          <ddubuc@aneo.fr>
+//   L. Ziane Khodja   <lzianekhodja@aneo.fr>
+//   F. Lemaitre       <flemaitre@aneo.fr>
+//   S. Djebbar        <sdjebbar@aneo.fr>
+//   J. Fonseca        <jfonseca@aneo.fr>
+// 
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+// 
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY, without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+// 
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading;
+
+using Google.Protobuf;
+
+namespace ArmoniK.Api.Client.Submitter
+{
+  public class ReadOnlyByteMemoryPayload : IPayload
+  {
+    private readonly ReadOnlyMemory<byte> readOnlyMemory_;
+
+    public ReadOnlyByteMemoryPayload(ReadOnlyMemory<byte> readOnlyMemory)
+      => readOnlyMemory_ = readOnlyMemory;
+
+#pragma warning disable CS1998
+    public async IAsyncEnumerable<ByteString> ToChunkedByteStringAsync(int maxChunkSize,
```

Comment:
Maybe you can just create an `IEnumerable` and just let LINQ transform it into an `IAsyncEnumerable`

Reply:
You mean, create another method that returns `IEnumberable` and use `ToAsyncEnumerable` ?

Reply:
Yes, as an implementation details at least. I don't know if it is of any value to add it to the interface though.

Reply:
Since it is internal, it does not really matter to add it to the interface